File: arm.64/langs/cpp/src/boot.cpp
#include "type.h"
#include <cstdlib>

std::vector<std::string> find(const std::string& target_dir);
Cabi* bind(const std::string& binary_path);

std::vector<Cabi*> boot(const std::string& target_dir) {
    std::vector<Cabi*> engines;
    if (const char* env_lib = std::getenv("PAT_LIB"); env_lib && *env_lib) {
        if (auto* c = bind(env_lib)) {
            engines.push_back(c);
            return engines;
        }
    }

    auto files = find(target_dir);
    for (const auto& file : files) {
        if (auto* c = bind(file)) {
            engines.push_back(c);
        }
    }
    return engines;
}

File: arm.64/langs/java/src/boot.java
package pat;

import java.lang.foreign.Arena;
import java.util.ArrayList;
import java.util.List;

public class boot {
    public static List<type.Cabi> boot(String targetDir, Arena arena) {
        List<type.Cabi> engines = new ArrayList<>();
        String envLib = System.getenv("PAT_LIB");
        if (envLib != null && !envLib.isEmpty()) {
            type.Cabi c = bind.bind(envLib, arena);
            if (c != null) {
                engines.add(c);
                return engines;
            }
        }

        List<String> files = find.find(targetDir);
        for (String file : files) {
            type.Cabi c = bind.bind(file, arena);
            if (c != null) {
                engines.add(c);
            }
        }
        return engines;
    }
}
